src/js/binaryen.js-post.js

```diff
@@ -3059,13 +3059,15 @@ Module['getExpressionInfo'] = function(expr) {
   const id = Module['_BinaryenExpressionGetId'](expr);
   const type = Module['_BinaryenExpressionGetType'](expr);
   switch (id) {
-    case Module['BlockId']:
+    case Module['BlockId']: {
+      const name = Module['_BinaryenBlockGetName'](expr);
       return {
         'id': id,
         'type': type,
-        'name': UTF8ToString(Module['_BinaryenBlockGetName'](expr)),
+        'name': name ? UTF8ToString(name) : null,
```

Comment:
Perhaps we could have a `UTF8ToStringOrNull` helper for this pattern, but we can consider that separately.